.github/workflows/build-core.yml

```diff
@@ -44,50 +44,73 @@ jobs:
           github-repo: ${{ github.repository }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
 
-  check_changelog:
+  update-changelog:
     runs-on: ubuntu-latest
+    outputs:
+      was_updated: ${{ steps.check-change.outputs.change_detected }}
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v2
         with:
-            path: copy-repo
-            fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
+          ref: ${{ github.head_ref }}
 
-      - name: Get changed files
-        id: changed-files
+      - name: Extract changelog info
+        id: extract-changelog
         run: |
-            cd copy-repo
-            if ${{ github.event_name == 'pull_request' }}; then
-                echo "changed_files=$(git diff --name-only -r HEAD^1 HEAD | xargs)" >> $GITHUB_OUTPUT
-            else
-                echo "changed_files=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }} | xargs)" >> $GITHUB_OUTPUT
-            fi
-
-      - name: List changed files
-        id: set-flag
+          PR_DESCRIPTION="${{ github.event.pull_request.body }}"
+
+          # Check if "changelog:" exists in PR description
+          if [[ $PR_DESCRIPTION == *"CHANGELOG:"* ]]; then
+
+            # Extract text after "changelog:"
+            CHANGELOG_TEXT=$(echo $PR_DESCRIPTION | sed -n 's/.*CHANGELOG: \(.*\)/\1/p')
+
+            echo "Extracted changelog: $CHANGELOG_TEXT"
+            echo "::set-output name=changelog::$CHANGELOG_TEXT"
+          else
+            echo "No changelog information found in PR description please add the changelog note"
+            exit 1
+          fi
+
+      - name: Check PR body against changelog
         run: |
-            cd copy-repo
-            for file in ${{ steps.changed-files.outputs.changed_files }}; do
-                echo "$file was changed"
-                if [[ $file == "CHANGELOG.md" ]]
-                then
-                  echo "file-flag=true" >> $GITHUB_OUTPUT
-                  break;
-                else
-                  echo "file-flag=false" >> $GITHUB_OUTPUT
-                fi
-            done
-
-      - name: Check if CHANGELOG is Updated and Abort if not updated
-        if: steps.set-flag.outputs.file-flag != 'true'
+          if ! grep -Fq "${{ steps.extract-changelog.outputs.changelog }}" CHANGELOG.md; then
+            LINE_NUMBER=$(awk '/^##/{print NR; exit}' CHANGELOG.md)
```

Comment:
So looks like LINE_NUMBER will be set to the line number of the first occurrence of a line starting with ## in CHANGELOG.md
which looks good 👍

Reply:
However, this will fail when the first changelog to be added is a changelog for a new Zowe version. And I have no idea how you'd automate against that, is there something in your Github action that can tell you what Zowe version it is? This is good for now, but here's what could happen:
1. Zowe PR author forgets changelog in a PR for new Zowe release
2. Automation adds changelog
3. Changelog gets committed into CHANGELOG.md into wrong Zowe release section, PR author doesn't know. 

You need some way for the PR author to sign off on this. Is there a GitHub action thing where PR is blocked w/o changelog as to enforce it, but the action itself needs to be check somehow by author? Even as something as, if they are responsible for clicking the button, and after a message pops up somewhere that says "Changelog has been added automatically. I tried my best, but check if it's the correct Zowe version!"

Reply:
Another issue:
Forgive me if I'm wrong, my bash is poor. I don't this checks if a CHANGELOG update already exists? ie a user can have a CHANGELOG update in their commits, but this isn't checked for

This could be done via Git action by checking if a commit in the PR contains the words "CHANGELOG"/"changelog"